logrotate: cleanup legacy options

martinetd opened the pull request:
###### Things done

This is a follow-up PR of #162063 to remove the compatibility options. I'll rebase/mark it ready in ~6months ; this is just about reserving a PR number for comment in code.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Artturin commented:
actually, we can just do this right after the 22.05 release so in 2-3 months

Artturin commented:
You should've pulled before pushing, I'll push back the changes

Artturin commented:
> You should've pulled before pushing, I'll push back the changes

I don't remember why I commented this, might have been meant for a another pr :P

martinetd commented:
22.05 is out -- I've rebased this so we can get rid of the compat logrotate code as promised :)

Changes since previous version:
- I've had to fix old release notes (can't link to no-longer existing options, tests failed) and added a note for removal in new one.
- added `mkRemovedOptionModule` for config/extraConfig/paths


cc @aanderse @Artturin

aanderse commented:
Awesome! I think we should include an `example` for `settings` now, though.

martinetd committed:
logrotate: add services.logrotate.settings example

martinetd commented:
> Awesome! I think we should include an example for settings now, though.

Good point, i've added one:
```
        example = literalExpression ''
          {
            # global options
            header = {
              dateext = true;
            };
            # example custom files
            "/var/log/mylog.log" = {
              frequency = "daily";
              rotate = 3;
            };
            "multiple paths" = {
               files = [
                "/var/log/first*.log"
                "/var/log/second.log"
              ];
            };
          };
          '';
```
(and tested the example in my config to make sure)


Rebased to address merge conflict in release notes.

Also please note a comment in the original PR made me open #177106 a few days ago, should I just add that commit here and close the other one? (doesn't really matter either way)

aanderse commented:
That example looks good!

martinetd commented:
Had forgotten about this and thought it had been merged... I've updated the release notes to make it mergable again.

I also rebased this on top of #180778 which is a simple fix for when sandbox is off -- happy to take that out but figuring it'll be less work to merge it all in one go and close the other PR as it's almost trivial

martinetd committed:
logrotate service: cleanup deprecated options

martinetd commented:
I've rebased again to fix merge conflict -- would really appreciate if we could merge this, small conflicts keep poping up and it's taking me time for something I don't really care about.
If option removal is early, we should at least get the config example in so I can drop the cleanup until e.g. next release again and stop worrying about rebase at this point.

aanderse commented:
@Artturin I don't really have the capacity for this right now. I think this should be good... can you please double check and merge? :heart:

martinetd commented:
Sorry/thanks -- I know way too well how not having time is, this can take the time it needs to take I just wanted to make sure I'm not rebasing this every months forever. Finding time keeps getting more expensive...

Artturin commented:
Ofborg passes, let's merge

martinetd commented:
Thanks!
